src/main/cpp/cleaner/windows/Makefile.mingw

```diff
@@ -16,5 +14,18 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-# 
 
+TMPFLD = ../../../../../target/tmp/
+OFLD = ../../../../../target/cleaner/
+
+all: prepfolder cleaner.exe
+
+prepfolder:
+	mkdir -p $(TMPFLD)
+	mkdir -p $(OFLD)
+
+clean:
+	rm -f *.res *.exe *.dll
+
+cleaner.exe: src/main.c
+	x86_64-w64-mingw32-gcc -s -DARCHITECTURE=64 -Wl,--nxcompat -Wl,--dynamicbase -Wl,--no-seh -Wl,--no-insert-timestamp -mwindows src/main.c -o$(OFLD)cleaner.exe -static -lstdc++ -static-libstdc++ -static-libgcc
```

Comment:
Is it a conscious choice to drop support for 32-bit binaries?
(I hope 32-bit binaries can be avoided, but just wanted to know if it it a conscious choice or not, to my knowledge 32-bit JRE/JDKs are still in active development for Java 17, at least by Azul and Adoptium)

Reply:
even my raspery pi runs on a mainline 64 bit kernel :) I don't think NB should bother with 32bit anymore. There is also nothing in the CI pipeline which is testing anything on 32bit JVMs right now - it makes sense to me to concentrate on 64bit only for sake of simplicity and reduced test workload.

Reply:
Yeah, well it is not about the OS. It is about the JRE. It is not uncommon to see people execute their workload on a 32-bit JRE (even on a very modern, latest and greatest 64-bit Windows or Linux OS), because it gives less memory consumption. 

But those are ETL workloads. I don't know if anyone would find it beneficial to execute their NetBeans IDE or NetBeans Platform application in 32 bit?  

I think it would be great if 32-bit support could be dropped.   Maybe the Decision Committee is us ?   :-)

Reply:
> I think it would be great if 32-bit support could be dropped. Maybe the Decision Committee is us ? :-)

Yes, that would be dev@ :wink: Suggest bringing it up there.

Reply:
> Yeah, well it is not about the OS. It is about the JRE. It is not uncommon to see people execute their workload on a 32-bit JRE (even on a very modern, latest and greatest 64-bit Windows or Linux OS), because it gives less memory consumption.

this used to be the case long ago. Today, all JVMs/GCs have support for compressed opps (enabled by default). And if you really do care about footprint to such extend that it matters there would be native images. (and even raspis have GBs of memory, I run three containers with 64bit JVMs in them on an old 1gb model and have still memory left)

> But those are ETL workloads. I don't know if anyone would find it beneficial to execute their NetBeans IDE or NetBeans Platform application in 32 bit?

right

> Yes, that would be dev@ wink Suggest bringing it up there.

i would be curious if it even works today given that probably nobody is testing this scenario.

Reply:
> Yes, that would be dev@ 😉 Suggest bringing it up there.

Well, I think we can decide here as this only affects NBI, come to think of it.  While there may (or not!) be some far fetched reason to run the IDE/Platform in 32-bit mode, the same is not true for NBI. 

So, my conclusion, and after some thought: No one will care if 32 bit support is dropped from NBI.

Reply:
@lbruun it's no secret I'd like to replace NBI in its entirety anyway (so I'm in favour). But sometimes people surprise with how they're using the platform, or what they're relying on. Is support shipped now? If so, at least some mention on dev@ would be good.

Reply:
I only target x64 as it was done in the launcher repo this way. There is only one cleaner.exe in the zip ([nbi](https://netbeans.osuosl.org/binaries/2B892D9E648A792EDB0F705378E06F84C322E208-nbi.engine-external-binaries-11.zip)) that I cannot determine the architecture.

cleaner.exe and others are used in the harness/libs.nbi.engine project in netbeans main repo (meged in a jar) and used in sequence in the nbm-maven-harness that is used to generate installer linux/windows for people using nbm-maven plugin. (may later be replaced with nbpacker but needs coding 😄 )

Reply:
> I only target x64 as it was done in the launcher repo this way. There is only one cleaner.exe in the zip ([nbi](https://netbeans.osuosl.org/binaries/2B892D9E648A792EDB0F705378E06F84C322E208-nbi.engine-external-binaries-11.zip)) that I cannot determine the architecture.

I've downloaded that existing ZIP in order to determine the bit-size of the `cleaner.exe` within that ZIP. The result:

```text
C:\Users\lars\Downloads>c:\tools\Microsoft-SysInternals\sigcheck.exe cleaner.exe

Sigcheck v2.73 - File version and signature viewer
Copyright (C) 2004-2019 Mark Russinovich
Sysinternals - www.sysinternals.com

C:\Users\lars\Downloads\cleaner.exe:
        Verified:       Unsigned
        Link date:      13:42 14-01-2010
        Publisher:      n/a
        Company:        n/a
        Description:    n/a
        Product:        n/a
        Prod version:   n/a
        File version:   n/a
        MachineType:    32-bit
```

It is 32-bit !!  
(not sure, that it matters, though)